svc/auth/endpoint.go

```diff
@@ -137,6 +148,12 @@ type (
 	UpdateUsernameRequest struct {
 		Username string `json:"username" validate:"required"`
 	}
+
+	// WhitelistRequest struct
+	WhitelistRequest struct {
+		AllowedType  string `json:"allowed_type,omitempty" validate:"required"`
```

Comment:
pls add validation rule for allowed type: 
```suggestion
		AllowedType string `json:"allowed_type,omitempty" validate:"required,oneof=email email_domain"`
```